financial-connections/src/main/java/com/stripe/android/financialconnections/repository/FinancialConnectionsConsumerSessionRepository.kt

```diff
@@ -145,17 +156,49 @@ private class FinancialConnectionsConsumerSessionRepositoryImpl(
         country: String,
     ): ConsumerSessionSignup = mutex.withLock {
         consumersApiService.signUp(
-            email = email,
-            phoneNumber = phoneNumber,
-            country = country,
-            name = null,
-            locale = locale,
-            amount = elementsSessionContext?.amount,
-            currency = elementsSessionContext?.currency,
-            incentiveEligibilitySession = elementsSessionContext?.incentiveEligibilitySession,
+            SignUpParams(
+                email = email,
+                phoneNumber = phoneNumber,
+                country = country,
+                name = null,
+                locale = locale,
+                amount = elementsSessionContext?.amount,
+                currency = elementsSessionContext?.currency,
+                incentiveEligibilitySession = elementsSessionContext?.incentiveEligibilitySession,
+                requestSurface = requestSurface,
+                consentAction = EnteredPhoneNumberClickedSaveToLink,
+                verificationToken = null,
+                appId = null,
+            ),
+            requestOptions = provideApiRequestOptions(useConsumerPublishableKey = false)
+        ).onSuccess { signup ->
+            updateCachedConsumerSessionFromSignup(signup)
+        }.getOrThrow()
+    }
+
+    override suspend fun mobileSignUp(
```

Comment:
Can we make both `signUp` methods call some internal method, so that we can avoid the code duplication?

Reply:
added `performSignUp`!